peer.go

```diff
@@ -1087,142 +1155,160 @@ func (p *peer) handleInitClosingSigned(req *htlcswitch.ChanClose, msg *lnwire.Cl
 			// channel closure.
-			req.Updates <- &lnrpc.CloseStatusUpdate{
-				Update: &lnrpc.CloseStatusUpdate_ChanClose{
-					ChanClose: &lnrpc.ChannelCloseUpdate{
-						ClosingTxid: closingTxid[:],
-						Success:     true,
+			if localReq != nil {
+				localReq.Updates <- &lnrpc.CloseStatusUpdate{
+					Update: &lnrpc.CloseStatusUpdate_ChanClose{
+						ChanClose: &lnrpc.ChannelCloseUpdate{
+							ClosingTxid: closingTxid[:],
+							Success:     true,
+						},
 					},
-				},
+				}
 			}
 		})
+	return nil, 0
 }
 
-// handleResponseClosingSigned is called when the responder in a cooperative
-// close workflow receives a ClosingSigned message. This function handles the
-// finalization of the cooperative close from the perspective of the responder.
-func (p *peer) handleResponseClosingSigned(msg *lnwire.ClosingSigned,
-	respSig []byte) {
-	p.activeChanMtx.RLock()
-	channel, ok := p.activeChannels[msg.ChannelID]
-	p.activeChanMtx.RUnlock()
-	if !ok {
-		peerLog.Errorf("unable to close channel, ChannelID(%v) is "+
-			"unknown", msg.ChannelID)
-		return
+// negotiateFeeAndCreateCloseTx takes into consideration the closing transaction
+// fee proposed by the remote peer in the ClosingSigned message and our
+// previously proposed fee (set to 0 if no previous), and continues the fee
+// negotiation it process. In case the peer agreed on the same fee as we
+// previously sent, it will assemble the close transaction and broadcast it. In
+// case the peer propose a fee different from our previous proposal, but that
+// can be accepted, a ClosingSigned message with the accepted fee is sent,
+// before the closing transaction is broadcasted. In the case where we cannot
+// accept the peer's proposed fee, a new fee proposal will be sent.
+//
+// TODO(halseth): In the case where we cannot accept the fee, and we cannot
+// make more proposals, this method should return an error, and we should fail
+// the channel.
+func (p *peer) negotiateFeeAndCreateCloseTx(channel *lnwallet.LightningChannel,
+	msg *lnwire.ClosingSigned, ourSig []byte,
+	ourFeeProp uint64) (*wire.MsgTx, []byte, uint64, error) {
+
+	peerFeeProposal := msg.FeeSatoshis
+
+	// If the fee proposed by the peer is different from what we proposed before
+	// (or we did not propose anything yet), we must check if we can accept the
+	// proposal, or if we should negotiate.
+	if peerFeeProposal != ourFeeProp {
+
+		// The peer has suggested a different fee from what we proposed.
+		// Let's calculate if this one is tolerable.
+		// TODO(halseth): use dynamic fee estimation.
+		ourIdealFeeRate := p.server.cc.feeEstimator.EstimateFeePerWeight(1) * 1000
+		ourIdealFee := channel.GetFee(ourIdealFeeRate)
+		fee := calculateCompromiseFee(ourIdealFee, ourFeeProp, peerFeeProposal)
+
+		// Our new proposed fee must be strictly between what we proposed before
+		// and what the peer proposed.
+		isStrictlyBetween := false
+		if fee < peerFeeProposal && fee > ourFeeProp {
```

Comment:
Not sure if it's more or less readable, but couldn't these two conditionals be combined with an ||?